python/ray/air/config.py

```diff
@@ -259,7 +259,7 @@ def from_placement_group_factory(
     ) -> "ScalingConfig":
         """Create a ScalingConfig from a Tune's PlacementGroupFactory"""
         if pgf.head_bundle_is_empty:
-            trainer_resources = {}
+            trainer_resources = None
```

Comment:
Hm I'm not fully familiar with the logic here, but are there other places in the code that might have expected this to be an empty dictionary?

Reply:
Actually, yeah, I believe this changes the logic and has to be reverted.

Reply:
I've moved the check.